src/benchmark.cpp

```diff
@@ -82,18 +261,16 @@ int main(int argc, char* argv[]) {
 #endif
 
         std::filesystem::path output_img_path_fname = entry.path().filename().replace_extension(".png");
-
         std::optional<cv::Mat> affine_transform;
-
+        BenchmarkGuardCSV benchmark_guard(entry.path().filename().string(), &benchmark_csv);
         {
-            BENCHMARK_BLOCK("parser");
-            affine_transform = run_parser("aruco", img,
+            affine_transform = run_parser(selected_parser, img,
 #ifdef DEBUG
                                           debug_img,
 #endif
                                           meta, dst_corner_points);
+            benchmark_guard.setSuccess(affine_transform.has_value());
```

Comment:
to do: mettre hors context setSuccess pour éviter la moindre influence sur les performance